services/search-repository.js

```diff
@@ -0,0 +1,66 @@
+const Search = require("../models/Search");
+
+addSearch = async ({ user, game, limit }) => {
+    const search = new Search({
+        user,
+        game,
+        limit,
+        party: [
+            {
+                user,
+            },
+        ],
+    });
+    await search.save();
+
+    return search;
+};
+
+deleteSearch = async ({ searchID }) => {
+    const search = await Search.findByIdAndDelete(searchID);
+    if (!search) {
+        throw new Error("NOT_FOUND");
+    }
+
+    return search;
+};
+
+joinSearch = async ({ searchID, playerID }) => {
+    const search = await Search.findById(searchID);
+    if (!search) {
+        throw new Error("NOT_FOUND");
+    }
+    if (search.party.some((player) => player.user === playerID)) {
+        throw new Error("ALREADY_EXISTS");
+    }
+    search.party.push({ user: playerID });
+    await search.save();
+    return search;
+};
+
+leaveSearch = async ({ searchID, playerID }) => {
+    const search = await Search.findById(searchID);
+    if (!search) {
+        throw new Error("NOT_FOUND");
+    }
+
+    if (!search.party.some((player) => player.user === playerID)) {
+        throw new Error("PLAYER_NOT_FOUND");
+    }
+
+    search.party.splice(
+        search.party.findIndex((player) => {
+            player.user === playerID;

```

Comment:
Can you verify if this works properly?

The callback `(player) => { player.user === playerID }` will always return `undefined` because you're not returning a value in the method body. The splice will not execute since the value returned by `findIndex` will always be `-1`.

For it to work, it should be either:

```js
(player) => {
  // notice the return?
  return player.user === playerID;
}
```

or

```js
(player) => player.user === playerID
// this is equivalent to the example above
```

To see the similarities/differences between the two examples, please refer to the [syntax documentation for arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#syntax).

Reply:
Addressed in a8aba9d2e6abbf19ec91ab9dea2919e2adba3c9c